src/ResourceManager/OperationalInsights/Commands.OperationalInsights/Query/InvokeOperationalInsightsQuery.cs

```diff
@@ -0,0 +1,100 @@
+﻿using Microsoft.Azure.Commands.Common.Authentication;
+using Microsoft.Azure.Commands.OperationalInsights.Models;
+using Microsoft.Azure.OperationalInsights;
+using Microsoft.Rest;
+using System;
+using System.Management.Automation;
+using System.Net;
+
+namespace Microsoft.Azure.Commands.OperationalInsights.Query
+{
+    [Cmdlet("Invoke", "AzureRmOperationalInsightsQuery"), OutputType(typeof(PSQueryResponse))]
+    public class InvokeOperationalInsightsQuery : ResourceManager.Common.AzureRMCmdlet
+    {
+        [Parameter(Mandatory = true, HelpMessage = "The workspace ID.")]
+        public string WorkspaceId { get; set; }
+
+        [Parameter(Mandatory = true, HelpMessage = "The query to execute.")]
+        public string Query { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = "The timespan to bound the query by.")]
```

Comment:
This seems rather unusual - is this necessary? If the user doesn't want to wait, they should create a powershell job I would think.

Reply:
The "Wait" here refers to how long the API should "wait" for the database to respond for a query. This is documented in the link that is included in the help message, but I'm open to renaming this parameter if you believe that will help.

Reply:
Okay we can talk about it when we do a cmdlet review.